### PR TITLE
specs updates for mmtracking mot components

### PR DESCRIPTION
**Pull request by yiyoulin**
Runs with latest pypi candidate (0.0.32a0) https://ml.azure.com/runs/1689a950-7577-400f-bd73-bece717b2065?wsid=/subscriptions/dbd697c3-ef40-488f-83e6-5ad4dfb78f9b/resourceGroups/vidani_rg/providers/Microsoft.MachineLearningServices/workspaces/vidani_wrk&tid=72f988bf-86f1-41af-91ab-2d7cd011db47#

### CODE REVIEW

File: assets/training/finetune_acft_image/components/finetune/mmt/spec.yaml [hunk 1]
```diff
@@ -0,0 +1,341 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandComponent.schema.json
+type: command
+
+version: 0.0.1
+name: mmtracking_video_multi_object_tracking_finetune
+display_name: Image Tracking MMTracking Model Finetune
```
Comment (PhaniShekhar) [inline assets/training/finetune_acft_image/components/finetune/mmt/spec.yaml:6]: display_name and description should be updated to reflect mot specific content.

File: assets/training/finetune_acft_image/environments/acft_video_mmtracking/context/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+azureml-metrics[video]====0.0.31
+azureml-evaluate-mlflow=={{latest-pypi-version}}
+azureml-acft-accelerator=={{latest-pypi-version}}
+azureml-acft-common-components[image]=={{latest-pypi-version}}
```
Comment (PhaniShekhar) [inline assets/training/finetune_acft_image/environments/acft_video_mmtracking/context/requirements.txt:4]: It looks like the pinning is in common-components for mmdetection environments. Could you please check.

File: assets/training/finetune_acft_image/environments/acft_video_mmtracking/context/requirements.txt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+azureml-acft-image-components==0.0.31
+azureml-core==1.52.0
+azure-ai-ml==1.5.0
+requests
+datasets==2.3.2
+optimum==1.6.4
+transformers==4.30.0
```
Comment (PhaniShekhar) [inline assets/training/finetune_acft_image/environments/acft_video_mmtracking/context/requirements.txt:11]: Please check if we can upgrade this and datasets to align with mmdetection environment.

Comment (PhaniShekhar) [reply assets/training/finetune_acft_image/environments/acft_video_mmtracking/context/requirements.txt:11]: transformers upgrade would also not result in vulnerabilities.

**Review by PhaniShekhar — CHANGES_REQUESTED** [review]

File: assets/training/finetune_acft_image/environments/acft_video_mmtracking/context/requirements.txt (hunk 3 above)
Comment (yiyoulin) [reply assets/training/finetune_acft_image/environments/acft_video_mmtracking/context/requirements.txt:11]: updated

File: assets/training/finetune_acft_image/components/finetune/mmt/spec.yaml (hunk 1 above)
Comment (yiyoulin) [reply assets/training/finetune_acft_image/components/finetune/mmt/spec.yaml]: changed. thanks for reminder.

File: assets/training/finetune_acft_image/environments/acft_video_mmtracking/asset.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+name: acft-mmtracking-video-gpu
```
Comment (rjaincc) [inline assets/training/finetune_acft_image/environments/acft_video_mmtracking/asset.yaml:1]: You would need one more PR for the env. 
@PhaniShekhar - had raised a PR to include the envs to a config.

File: assets/training/finetune_acft_image/components/finetune/mmt/spec.yaml [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+    optional: true
+    description: image width that is input to the network. Default 
+                 is -1 which means it would be overwritten by image_scale in model config. 
+
+  image_height:
+    type: integer
+    default: -1
+    optional: true
+    description: image height that is input to the network. Default 
+                 is -1 which means it would be overwritten by image_scale in model config.
```
Comment (rjaincc) [inline assets/training/finetune_acft_image/components/finetune/mmt/spec.yaml:50]: extra spaces - can we please clean those up?

File: assets/training/finetune_acft_image/components/finetune/mmt/spec.yaml [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+  $[[--save_steps ${{inputs.save_steps}}]]
+  $[[--save_total_limit ${{inputs.save_total_limit}}]]
+  $[[--apply_early_stopping ${{inputs.early_stopping}}]]
+  $[[--early_stopping_patience ${{inputs.early_stopping_patience}}]]
+  $$[[--max_grad_norm ${{inputs.max_grad_norm}}]]
+  $[[--resume_from_checkpoint ${{inputs.resume_from_checkpoint}}]]
+  $[[--save_as_mlflow_model ${{inputs.save_as_mlflow_model}}]]
+  --mlflow_model_folder ${{outputs.mlflow_model_folder}}
+  --pytorch_model_folder ${{outputs.pytorch_model_folder}}
+  --per_device_eval_batch_size 1
```
Comment (rjaincc) [inline assets/training/finetune_acft_image/components/finetune/mmt/spec.yaml:341]: Why hard code it, why not keep 1 as default in task specific config?

File: assets/training/finetune_acft_image/components/model_import/mmt/spec.yaml [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    enum:
+      - MmTrackingVideo
+    description: Which framework the model belongs to.
+
+  # Model name
+  model_name:
+    type: string
+    optional: true
+    description: Please select models from AzureML Model Assets for all supported models.
+     For MMTracking, provide the model's config name here, same as its specified in MMTracking Model Zoo.
```
Comment (rjaincc) [inline assets/training/finetune_acft_image/components/model_import/mmt/spec.yaml:31]: Extra spaces - same comment for other places in this file.

File: assets/training/finetune_acft_image/components/finetune/mmt/spec.yaml [hunk 8]
```diff
@@ -0,0 +1,341 @@
+$schema: https://azuremlschemas.azureedge.net/latest/commandComponent.schema.json
```
Comment (rjaincc) [inline assets/training/finetune_acft_image/components/finetune/mmt/spec.yaml:1]: Would you need any changes in the supporting finetune.py file?
If so, are you planning on pulling in those at a later point in time?

File: assets/training/finetune_acft_image/components/finetune/mmt/spec.yaml (hunk 5 above)
Comment (yiyoulin) [reply assets/training/finetune_acft_image/components/finetune/mmt/spec.yaml:50]: all updated accordingly. no more new lines !! thanks Rupal!

File: assets/training/finetune_acft_image/components/model_import/mmt/spec.yaml (hunk 7 above)
Comment (yiyoulin) [reply assets/training/finetune_acft_image/components/model_import/mmt/spec.yaml]: all updated accordingly. no more new lines !! thanks Rupal!

File: assets/training/finetune_acft_image/components/finetune/mmt/spec.yaml (hunk 8 above)
Comment (yiyoulin) [reply assets/training/finetune_acft_image/components/finetune/mmt/spec.yaml:1]: added. ;) thank you.

File: assets/training/finetune_acft_image/environments/acft_video_mmtracking/asset.yaml (hunk 4 above)
Comment (yiyoulin) [reply assets/training/finetune_acft_image/environments/acft_video_mmtracking/asset.yaml:1]: it turns out I don't have write permission to this PR... I'll bother Phani or other IDC folks to write for me :(

Comment (rjaincc) [reply assets/training/finetune_acft_image/environments/acft_video_mmtracking/asset.yaml:1]: Here is the PR https://github.com/microsoft/mcr/pull/2892

File: assets/training/finetune_acft_image/environments/acft_video_mmtracking/context/requirements.txt [hunk 9]
```diff
@@ -1,14 +1,14 @@
-azureml-metrics[video]====0.0.31
+azureml-metrics[image]==0.0.32
```
Comment (PhaniShekhar) [inline assets/training/finetune_acft_image/environments/acft_video_mmtracking/context/requirements.txt:1]: Should this not be azureml-metrics[video] ?

**Review by PhaniShekhar — APPROVED** [review]

**Pull request merged**